src/views/Project/index.jsx

```diff
@@ -155,6 +156,23 @@ export default class Project extends Component {
       this.load();
     }
   }
+
+  handleFetchBugFirstComment = async id => {
+    try {
+      const {
+        data: { comments },
+      } = await this.props.client.query({
+        query: commentsQuery,
+        variables: { id },
+        context: { link: 'bugzilla' },
+      });
+
+      return comments[0].text;
+    } catch (err) {
+      return '';
```

Comment:
This doesn't seem to do anything with the error. If anything goes wrong, we can handle it using `componentDidCatch`. In tc-web, it's done in https://github.com/taskcluster/taskcluster-web/blob/2998f6561a4de26908d4a2eba4aaca8443c15a53/src/App/index.jsx#L79-L81. Once the error state is set, `App/index.jsx` displays the ErrorPanel.

Reply:
I tried putting `componentDidCatch` in the Project/index.jsx then i change the query comments to be commentzz which will return error (ApolloError), but it only show the error in the console instead of the panel, then I move it to the App/index.jsx and it still show only in the console. is it because the error came from event handler? https://reactjs.org/docs/error-boundaries.html#how-about-event-handlers 

i agree the catch block here did nothing so i added setState in the catch section

Reply:
Did you make sure to remove the try and catch statement before running the query again?

Reply:
yes